DEVELOPMENT.md

```diff
@@ -429,12 +406,71 @@ git submodule update --init --remote
 git submodule foreach "git fetch --all"
 ```
 
-Then run tests as usual: `cargo test`.
+Then run tests as usual: `cargo test --all-features --all-targets`.
+
+We strive for two kind of tests, the classic unit tests inside of the
+implementation files. Additionally we have a suit of integration tests esp. to
+assert the API the proxy provides from a high-level consumer perspective, these
+can be found under `proxy/tests`. To find out where to place and how to layout
+tests check Rust book [test chapter][rt].
+
+### File structure
+
+We understand the GraphQL API as one potential surface to expose the
+Application domain logic. Therefore we try to treat it as a thin layer exposing
+well-typed entities. The heavy lifting is done in the modules named after the
+protocols we consume - `CoCo` for code collaboration and `Registry` for global
+unique entries for users, projects and organisations. By isolating concerns
+this way, we hope to enable ease-of-contribution to downstream teams.
+Empowering them to reflect changes in their public APIs easily with code
+contributions to Upstream.
+
+```
+proxy/src/
+├── coco.rs
+├── env.rs
+├── error.rs
+├── graphql
+│   ├── api.rs
+│   ├── error.rs
+│   ├── project.rs
+│   └── schema.rs
+├── graphql.rs
+├── lib.rs
+├── main.rs
+└── registry.rs
+```
 
 💡 *You'll have to run the submodule commands every time there are any updates
 to the test fixture repository.*
```

Comment:
This should be moved back to the `testing` section below the `git submodule` commands (around L:408).